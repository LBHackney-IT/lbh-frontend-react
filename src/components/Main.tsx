import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import { Attributes, DataAttributes } from "../helpers/Attributes";

import "lbh-frontend/lbh/objects/_main-wrapper.scss";

/**
 * The proptypes for {@link Main}.
 *
 * This also supports all `aria-*` and `data-*` attributes.
 *
 * @noInheritDoc
 */
export interface MainProps extends React.AriaAttributes, DataAttributes {
  /**
   * Defaults to `"main-content"` if not set.
   */
  id?: string;
  className?: string;
  /**
   * The elements within the main wrapper. Contains the content you deem to be
   * the most important for your page.
   */
  children: React.ReactNode;
}

/**
 * A wrapper for the main content within the page. Can be used as an indicator
 * of where the main content begins within a page, so somebody using assistive
 * technology can navigate to the information that they need more easily.
 */
export const Main: React.FunctionComponent<MainProps> = (
  props: MainProps
): JSX.Element => {
  const { id, className, children } = props;

  const extraAttributes = Attributes.ariaAndData(props);

  return (
    <main
      className={classNames(
        "govuk-main-wrapper",
        "lbh-main-wrapper",
        className
      )}
      id={id ? id : "main-content"}
      role="main"
      {...extraAttributes}
    >
      {children}
    </main>
  );
};

Main.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};
