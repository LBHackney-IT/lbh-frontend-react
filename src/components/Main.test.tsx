import React from "react";
import { create } from "react-test-renderer";

import { Main } from "./Main";

it("renders correctly with all props", () => {
  const component = create(
    <Main id="test" className="class1 class2">
      Test
    </Main>
  );

  expect(component).toMatchInlineSnapshot(`
    <main
      className="govuk-main-wrapper lbh-main-wrapper class1 class2"
      id="test"
      role="main"
    >
      Test
    </main>
  `);
});

it("renders correctly without optional props", () => {
  const component = create(<Main>Test</Main>);

  expect(component).toMatchInlineSnapshot(`
    <main
      className="govuk-main-wrapper lbh-main-wrapper"
      id="main-content"
      role="main"
    >
      Test
    </main>
  `);
});
