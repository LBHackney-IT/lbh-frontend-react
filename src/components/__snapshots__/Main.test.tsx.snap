// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<main
  className="govuk-main-wrapper lbh-main-wrapper class1 class2"
  id="test"
  role="main"
>
  Test
</main>
`;

exports[`renders correctly without optional props 1`] = `
<main
  className="govuk-main-wrapper lbh-main-wrapper"
  id="main-content"
  role="main"
>
  Test
</main>
`;
